src/rgw/rgw_rest_pubsub.cc

```diff
@@ -885,24 +895,32 @@ void RGWPSSetTopicAttributesOp::execute(optional_yield y) {
     // initialize the persistent queue's location, using ':' as the namespace
     // delimiter because its inclusion in a TopicName would break ARNs
     dest.persistent_queue = string_cat_reserve(topic_arn.account, ":", topic_name);
-
-    op_ret = driver->add_persistent_topic(this, y, dest.persistent_queue);
-    if (op_ret < 0) {
-      ldpp_dout(this, 4)
-          << "SetTopicAttributes Action failed to create queue for "
-             "persistent topics. error:"
-          << op_ret << dendl;
-      return;
+    dest.num_shards = s->get_cct()->_conf.get_val<uint64_t>("rgw_bucket_persistent_notif_num_shards");
+    for(const auto& q: dest.get_queue_names()){
+      op_ret = driver->add_persistent_topic(this, y, q);
```

Comment:
same doubt here as raised in the previous comment. 

assume in a mixed cluster,  i create a persistent topic `t1` on a newer rgw, and set it to be non-persistent from an older rgw - this would only delete the first shard. 

if i make it persistent again from the newer rgw, delete and recreate or reuse the same existing shards.